pysb/core.py

```diff
@@ -1444,7 +1460,7 @@ def enable_synth_deg(self):
         source_cp = as_complex_pattern(self.monomers['__source']())
         if self.compartments:
             for c in self.compartments:
-                source_cp = source_cp ** c
+                source_cp.compartment = c
```

Comment:
Just avoiding usage of the overloaded operators within the core? I thought that might be a good style policy but it's not written down anywhere.